### DISCUSSION TITLE
Help to use the ASCII EPA Airnow data into METplus

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1515] [File I/O] [answered]
Hi, there,

I had been mentioning that the EPA will begin to provide ASCII files of AIRNOW data rather than the bufr files they send us now.  Currently, METplus uses PB2NC to read in a prepbufr file generated from the bufr tanks the EPA sends over, but they will start to send us only ASCII files, and we'd like your assistance in being able to read them.

You can find sample files on WCOSS Venus here:  /gpfs/dell2/emc/modeling/noscrub/Ho-Chun.Huang/epa_airnow_acsii/2022/20220312

There are several files here to look at, but they both provide the same data.  The file with all the data is the HourlyAQObs data.  The HourlyData is a neater looking file with the same data, but the lat/lon of each station is not provided there, so you'll have to use the HourlyAQObs file, which is a CSV file.

This file should allow you to learn what is in each column of the CSV file.  
[HourlyAQObsFactSheet.pdf](https://github.com/dtcenter/METplus/files/8256941/HourlyAQObsFactSheet.pdf)

This file you can look at for the HourlyData file, in case you can use that.
[HourlyDataFactSheet.pdf](https://github.com/dtcenter/METplus/files/8256946/HourlyDataFactSheet.pdf)

It would be great to be able to also read in the daily file, which I think we can use to validate daily model items.  That is the daily_data.dat file, and there are two of them.  Unfortunately, they are in the HourlyData format, but you should be able to use the lat/lon of the HourlyAQObs file to find these stations.    Here's the fact sheet on the daily file: 

[HourlyDataFactSheet.pdf](https://github.com/dtcenter/METplus/files/8256977/HourlyDataFactSheet.pdf)

As you can see, they don't make this easy.  However, we could just stick with hourly data and just sum/average them the way we are doing now in PB2NC.  Let me know if there are any questions regarding this file.

Thanks!

Perry

**Comment by DanielAdriaansen** [comment, marked as answer]
Hello Perry,

Please follow the instructions here for sending us some sample data:
https://github.com/dtcenter/METplus/discussions/954

Could you log on to WCOSS and collect some relevant sample data for this work and send it via FTP? Please include all types of files you think we will need to be able to read. Thanks!

-Dan

**Reply by PerryShafran-NOAA** [reply]
Files are in the shafran_data directory on your ftp server.

Perry

**Reply by DanielAdriaansen** [reply]
@PerryShafran-NOAA can you describe the difference here:

> That is the daily_data.dat file, and there are two of them.

I see:
```
daily_data.data
daily_data_v2.dat
```

Also, is the hourly fact sheet the same for the hourly and the daily files? In your original post you linked to the Hourly fact sheet attachment where you are referring to the daily files:

> Here's the fact sheet on the daily file:
> 
> [HourlyDataFactSheet.pdf](https://github.com/dtcenter/METplus/files/8256977/HourlyDataFactSheet.pdf)

Thank you!

**Reply by PerryShafran-NOAA** [reply]
I do believe that I sent you the wrong sheet for the daily file.   This one is the correct sheet for the daily file.  My apologies for sending the wrong one previously.  

[DailyDataFactSheet.pdf](https://github.com/dtcenter/METplus/files/8288703/DailyDataFactSheet.pdf)

Perry

**Reply by DanielAdriaansen** [reply]
I only see `daily_data_v2.dat` described in the fact sheet. Do you have any information about the `daily_data.dat` file?

**Reply by PerryShafran-NOAA** [reply]
I thought that the DailyDataFactSheet applied to both, but maybe not?  It seems that the format of both files are the same.  Let's assume that they are and use the fact sheet for both.  I'm not entirely sure what's the difference between the v2 and the non-v2 file, so you might need to compare the two files.

**Reply by DanielAdriaansen** [reply]
We have opened an issue for this work here: https://github.com/dtcenter/MET/issues/2142. Please feel free to join the conversation on that issue and provide and further comments there. Thanks Perry!

**Reply by PerryShafran-NOAA** [reply]
Thank you so much for the update!
